dot/peerset/peerset_test.go

```diff
@@ -113,73 +211,186 @@ func TestPeerSetDiscovered(t *testing.T) {
 
 func TestReAllocAfterBanned(t *testing.T) {
 	t.Parallel()
-
 	handler := newTestPeerSet(t, 25, 25, []peer.ID{}, []peer.ID{}, false)
 
 	ps := handler.peerSet
-	// adding peer1 with incoming slot.
-	if ps.peerState.peerStatus(0, peer1) == unknownPeer {
-		ps.peerState.discover(0, peer1)
-		err := ps.peerState.tryAcceptIncoming(0, peer1)
-		require.NoError(t, err)
-	}
+	require.Equal(t, unknownPeer, ps.peerState.peerStatus(testSetID, peer1))
+
+	ps.peerState.discover(testSetID, peer1)
+	err := ps.peerState.tryAcceptIncoming(testSetID, peer1)
+	require.NoError(t, err)
+
+	// accepting the income peer which is not in the reserved peers
+	// should increase the numIn count by 1
+	checkPeerStateSetNumIn(t, ps.peerState, testSetID, 1)
 
 	// We ban a node by setting its reputation under the threshold.
 	rep := newReputationChange(BannedThresholdValue-1, "")
-	// we need one for the message to be processed.
 	handler.ReportPeer(rep, peer1)
-	time.Sleep(time.Millisecond * 100)
 
+	time.Sleep(time.Millisecond * 100)
 	checkMessageStatus(t, <-ps.resultMsgCh, Drop)
 
-	// Check that an incoming connection from that node gets refused.
+	// banning a incoming peer should decrease the numIn count by 1
+	checkPeerStateSetNumIn(t, ps.peerState, testSetID, 0)
+	checkNodePeerMembershipState(t, ps.peerState, peer1, testSetID, notConnected)
 
-	handler.Incoming(0, peer1)
+	n, exists := getNodePeer(t, ps.peerState, peer1)
+	require.True(t, exists)
+
+	// when the peer1 was banned we updated its lastConnected field to time.Now()
+	lastTimeConnected := n.lastConnected[testSetID]
+
+	// Check that an incoming connection from that node gets refused.
+	handler.Incoming(testSetID, peer1)
 	checkMessageStatus(t, <-ps.resultMsgCh, Reject)
 
-	time.Sleep(time.Millisecond * 100)
+	// when calling Incoming method the peer1 is with status notConnectedPeer
+	// so we update its lastConnected field to time.Now() again
+	n, exists = getNodePeer(t, ps.peerState, peer1)
+	require.True(t, exists)
+
+	currentLastTimeConnected := n.lastConnected[testSetID]
+	require.True(t, lastTimeConnected.Before(currentLastTimeConnected))
+
+	// wait a bit for the node's reputation to go above the threshold.
+	<-time.After(allocTimeDuration + time.Second)
```

Comment:
nit same thing really

```suggestion
	time.Sleep(allocTimeDuration + time.Second)
```